content/developer/reference/frontend/error_handling.rst

```diff
@@ -0,0 +1,581 @@
+    }
+    // start working with someVal here
+
+While this is straightforward with try/catch, it's much easier to accidentally wrap
+a much larger portion of code in a catch clause when working with `Promise.catch`:
+
+.. code-block:: javascript
+
+    someFunction().then((someVal) => {
+      // work with someVal
+    }).catch((e) => {
+      if (!(e instanceof MyError)) {
+        throw e;
+      }
+      return null;
+    });
+
+In this example, the catch block is actually catching errors in the entire then
+block, which is not what we want. In this particular example, because we properly
+filter based on the error type, we're not swallowing the error, but you can see
+that it may be much easier to do so if we're expecting a single error type and decide
+not to have the instanceof check. Notice however that unlike the previous example,
+the null isn't going through the codepath that uses `someVal`. To avoid this,
+catch clauses should generally be as close as possible to the promise that may throw,
+and should always filter on the error type.
+
+Error free control flow
+=======================
+
+For the reasons outlined above, you should avoid throwing errors for doing routine
+things, and in particular, for control flow. If a function is expected to be unable
+to complete its work on a regular basis, it should communicate that failure without
+throwing an exception. Consider the example code:
+
+.. code-block:: javascript
+
+    let someVal;
+    try {
+      someVal = someFunction();
+    } catch (e) {
+      if (!(e instanceof MyError)) {
+        throw e;
+      }
+      someVal = null;
+    }
+
+There are many things that are problematic with this code. First, because we want
+the variable `someVal` to be accessible after the `try/catch` block, it needs to be
+declared before that block, and it cannot be `const` since it needs to be assigned
+after initialization. This hurts readability further down the road as you now have
+to look out for this variable potentially being reassigned later in the code.
+
+Second, when we catch the error, we have to check that the error is actually the type
+of error we were expecting to catch, and if not, rethrow the error. If we don't do
+this, we might end up swallowing errors that were *actually* unexepected instead of
+reporting them correctly, e.g. we could be catching and swallowing a TypeError if the
+underlying code tries to access a property on `null` or `undefined`.
+
+Lastly, not only is this very verbose, but it's easy to do this incorrectly: if you
+forget to add the `try/catch`, you will likely end up with a traceback. If you add
+the `try/catch` block but forget to rethrow unexpected errors, you will end up swallowing
+unrelated errors. And if you want to avoid having to reassign the variable you may
+move the entire block that uses the variable inside the `try` block. The more code
+you have inside your `try` block, the more likely you are to catch unrelated errors,
```

Comment:
also, more code there is in the try block, less obvious it is which instruction may throw the expected error